kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/sequences/matchers.kt

```diff
@@ -278,12 +278,20 @@ infix fun <T> Sequence<T>.shouldNotHaveSingleElement(t: T) = this shouldNot sing
 
 fun <T> singleElement(t: T) = object : Matcher<Sequence<T>> {
    override fun test(value: Sequence<T>): MatcherResult {
-      val valueAsList = value.toList()
-      val actualCount = valueAsList.count()
+      var failureMessage: String? = null
+      val iterator = value.iterator()
+      var currentValue: T?
```

Comment:
Now that we have expected and actual elements and sequences, can we improve naming precision? Like `t` -> `expectedElement`, `currentValue` -> `actualElement`. (Unfortunately, renaming `value` would conflict with the base function signature.)

Reply:
Yeah, of course. Good point!